crates/cadenza-eval/src/compiler.rs

```diff
@@ -0,0 +1,120 @@
+//! Compiler API surface for the evaluator.
+//!
+//! The `Compiler` struct is the shared mutable state that accumulates
+//! definitions during evaluation. Macros call back into the compiler
+//! API to register definitions, emit IR, etc.
+
+use crate::interner::InternedId;
+use crate::value::Value;
+use std::collections::HashMap;
```

Comment:
We should use a crate type alias for maps. The std hashmap has DoS protection which we don't want to pay for a local compiler. So something like

```rust
// In interner.rs
pub type Map<V> = std::collections::HashMap<InternedId, V, FxHash>;
```